"""
This is a test file for testing `lazy_matrix_mul` function with doctest.

To run these tests, use the following command:
python3 -m doctest -v tests/101-lazy_matrix_mul.txt
"""

def lazy_matrix_mul(m_a, m_b):
    """
    >>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
    array([[13, 16]])

    >>> lazy_matrix_mul([[1]], [[2]])
    array([[2]])

    >>> lazy_matrix_mul([[1, 2, 3], [4, 5, 6]], [[7, 8], [9, 10], [11, 12]])
    array([[ 58,  64],
           [139, 154]])

    >>> lazy_matrix_mul([[1, 2]], [[1], [2]])
    array([[5]])

    >>> lazy_matrix_mul([], [])
    Traceback (most recent call last):
    ...
    ValueError: ...

    >>> lazy_matrix_mul([[]], [[]])
    Traceback (most recent call last):
    ...
    ValueError: ...

    >>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2, 3], [4, 5, 6]])
    Traceback (most recent call last):
    ...
    ValueError: ...

    >>> lazy_matrix_mul("not a list", [[1, 2], [3, 4]])
    Traceback (most recent call last):
    ...
    TypeError: ...

    >>> lazy_matrix_mul([[1, 2], [3, 4]], "not a list")
    Traceback (most recent call last):
    ...
    TypeError: ...

    >>> lazy_matrix_mul([[1, 2], [3, "four"]], [[1, 2], [3, 4]])
    Traceback (most recent call last):
    ...
    TypeError: ...

    >>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], ["three", 4]])
    Traceback (most recent call last):
    ...
    TypeError: ...

    >>> lazy_matrix_mul([[1, 2, 3]], [[4], [5], [6]])
    array([[32]])

    >>> lazy_matrix_mul([[1, 2, 3], [4, 5, 6]], [[7], [8], [9]])
    Traceback (most recent call last):
    ...
    ValueError: ...
    ""‚Äù
